From dfb38a7c0168d32ee84a76bf6e2c0dff75cf77d8 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Wed, 13 Dec 2017 22:38:52 +0000
Subject: [PATCH 3/3] Add documentation to man page for kernel configuration

Update man page to describe the use of the kernel configuration files
that can be modified by the system administrator to set the kernel
commandline.
---
 man/{clr-boot-manager.1 => clr-boot-manager.1.in} | 13 +++++++++++++
 man/meson.build                                   | 10 +++++++++-
 2 files changed, 22 insertions(+), 1 deletion(-)
 rename man/{clr-boot-manager.1 => clr-boot-manager.1.in} (90%)

diff --git a/man/clr-boot-manager.1 b/man/clr-boot-manager.1.in
similarity index 90%
rename from man/clr-boot-manager.1
rename to man/clr-boot-manager.1.in
index ed76d2b..e0900f0 100644
--- a/man/clr-boot-manager.1
+++ b/man/clr-boot-manager.1.in
@@ -93,6 +93,19 @@ Returns the currently configured timeout as set by \fBclr\-boot\-manager set\-ti
 .PP
 On success, 0 is returned, a non\-zero failure code otherwise\&
 
+.SH "CONFIGURATION"
+.PP
+clr-boot-manager takes into account the following paths to modify its behavior:
+
+.PP
+\fB@KERNEL_CONF_DIRECTORY@/cmdline.d/*.conf\fR
+.RS 4
+A set of files that will be used to modify the kernel commandline. The files
+can also be used to mask the vendor cmdline if the filename matches a vendor
+configuration file and is linked to /dev/null\&.
+.RE
+
+
 .SH "ENVIRONMENT"
 \fI$CBM_DEBUG\fR
 .RS 4
diff --git a/man/meson.build b/man/meson.build
index 08c1136..b71caf4 100644
--- a/man/meson.build
+++ b/man/meson.build
@@ -1,6 +1,14 @@
 mandir = join_paths(path_prefix, get_option('mandir'), 'man1')
 
+# Update man page with paths
+man_data = configuration_data()
+man_data.set('KERNEL_CONF_DIRECTORY', with_kernel_conf_dir)
+man_1 = configure_file(input : 'clr-boot-manager.1.in',
+                       output : 'clr-boot-manager.1',
+                       configuration : man_data,
+)
+
 install_data(
-    'clr-boot-manager.1',
+    man_1,
     install_dir: mandir,
 )
-- 
2.15.1

